### ISSUE TITLE
HTTP 415 when syncing CalDav on Nextcloud with --init=keep-remote

### ISSUE THREAD

**Issue by cryptochimp** [lfos/calcurse#133]
When running `calcurse-caldav --debug --init=keep-remote` I get a 415 reply (complete output below). 

`calcurse-caldav --debug --init=keep-remote` output
```
warning: Dry run; nothing is imported/exported. Add "DryRun = No" to the
warning: [General] section in the configuration file to enable synchronization.
Connecting to [HOST]...
Removing all local calcurse objects...
> REPORT [PATH]
> Headers: {'Authorization': 'Basic [TOKEN]', 'Content-Type': 'application/xml; charset=utf-8'}
> <?xml version="1.0" encoding="utf-8" ?><C:calendar-query xmlns:D="DAV:"                   xmlns:C="urn:ietf:params:xml:ns:caldav"><D:prop><D:getetag /></D:prop><C:filter><C:comp-filter name="VCALENDAR" /></C:filter></C:calendar-query>

< Headers: [('Date', 'Mon, 30 Jul 2018 08:56:04 GMT'), ('Server', 'Apache'), ('Expires', 'Thu, 19 Nov 1981 08:52:00 GMT'), ('Cache-Control', 'no-store, no-cache, must-revalidate'), ('Pragma', 'no-cache'), ('Content-Security-Policy', "default-src 'none';"), ('Set-Cookie', 'oc_sessionPassphrase=[TOKEN]; path=[PATH]; secure; HttpOnly'), ('Set-Cookie', '[TOKEN]; path=[PATH]; secure; HttpOnly'), ('Set-Cookie', 'cookie_test=test; expires=Mon, 30-Jul-2018 09:56:05 GMT; Max-Age=3600'), ('X-Content-Type-Options', 'nosniff'), ('X-XSS-Protection', '1; mode=block'), ('X-Robots-Tag', 'none'), ('X-Frame-Options', 'SAMEORIGIN'), ('X-Download-Options', 'noopen'), ('X-Permitted-Cross-Domain-Policies', 'none'), ('Transfer-Encoding', 'chunked'), ('Content-Type', 'application/xml; charset=utf-8')]
< <?xml version="1.0" encoding="utf-8"?>
< <d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
<   <s:exception>Sabre\DAV\Exception\ReportNotSupported</s:exception>
<   <s:message/>
<   <d:supported-report/>
< </d:error>

error: The server at [HOST] replied with HTTP status code 415
error: (Unsupported Media Type) while trying to access
error: [PATH].
```

config file
```
[General]
Binary = calcurse
Hostname = [HOST]
Path = [PATH]
AuthMethod = basic
InsecureSSL = No
SyncFilter = cal,todo
DryRun = Yes
Verbose = Yes

[Auth]
Username = [USERNAME]
Password = [PASSWORD]
```

Looking around it seems that a possible cause might be the `Content-Type` header. However when I write `Content-Type = application/json` after `[CustomHeaders]` in the config file, calcurse sends BOTH headers `content-type=application/json` and `Content-Type=application/xml; charset=utf-8`. So it seems the header in the config file is transformed to lower case and I cannot overwrite the Content-Type. In any case, I don't really know if this is one solution but I couldn't find more info online.

**Comment by loweryaustin** [comment]
The issue is not with the content type header. Where did you get the idea that it would be JSON rather than XML? I ask only because I am pretty sure that the DAV protocol uses xml rather than JSON, but I don't know enough to be positive.

The core of the issue is the following error:
`<   <s:exception>Sabre\DAV\Exception\ReportNotSupported</s:exception>`
This link might have an extra clue about the issue:
https://www.apt-browse.org/browse/ubuntu/trusty/universe/all/php-sabre-dav/1.7.6+dfsg-2/file/usr/share/php/Sabre/DAV/Exception/ReportNotSupported.php

But you're redacted the REPORT part of the output as far as I can tell. Does this have to do with the fact that you are doing a dry run? I wonder if NextCloud does not support the dry run. Just throwing ideas out there.

I did a bit of digging and found the code that generates this error:
https://github.com/sabre-io/dav/blob/master/lib/DAV/Exception/ReportNotSupported.php
` * This exception is thrown when the client requested an unknown report through the REPORT method`
That makes me more curious about what the redacted REPORT command is.

**Comment by cryptochimp** [comment]
Thanks for the reply. I found a couple of reports in which an app would send the incorrect content type and that's why the server was replying with a 415, so I tried to change that and I realized that it didn't work because of the upper/lower case problem. But in any case, as you mention that is probably not the problem, it seems to be a communication error rather than a bug here so I'm closing this. I will try to find more about this and comment in case I find something interesting.
